Feature: turn style guide into brand manual

madcampos opened the pull request:
## Summary
The objective of this rewrite is to turn the style guide into a brand manual that can be re-expanded into a new set of guidelines and components for creating branded content. It also offers some templates for banners and logos.

## Changelog
- Move project to vite.
- Move complex logic to web components.
- Add logos and templates as SVGs for portability and editability.
- Add sections for brand colors
- Add section for brand fonts

madcampos committed:
feat: add project files

madcampos committed:
fix: resolve misc issues from migration

madcampos committed:
feat: update config and package details

madcampos committed:
feat: move assets to root assets folder

madcampos committed:
feat: add basic css

madcampos committed:
feat: add partial content and structure

madcampos committed:
feat: add fonts

madcampos committed:
feat: add text styles

madcampos committed:
feat: update colors css

madcampos committed:
chore: remove old files

madcampos committed:
feat: add toc code

madcampos committed:
feat: add logo section

madcampos committed:
fix: make paths relative

madcampos committed:
fix: update colours and typography sections

madcampos committed:
fix: leftover tweaks

madcampos committed:
chore: remove .DS_Store files

madcampos committed:
feat: add events images

madcampos committed:
chore: remove old code and move code around

madcampos committed:
feat: add download option to logo custom element

madcampos committed:
feat: add graphic material section

madcampos committed:
feat: add graphic material images

madcampos committed:
feat: add pictures section

madcampos committed:
feat: add image download to logos

madcampos committed:
feat: update toc links

madcampos committed:
feat: update build target

madcampos committed:
feat: add built version

madcampos committed:
feat: add build ignore for github

madcampos committed:
feat: ignore build

madcampos committed:
feat: update urls

madcampos committed:
feat: update fonts location

madcampos committed:
chore: remove unused files

madcampos committed:
feat: update built project

madcampos committed:
feat: add color boxes

madcampos committed:
feat: reference new constants file

madcampos committed:
fix: remove extra filler text

madcampos committed:
fix: update color card and color card container css

madcampos committed:
feat: add dimnesions to the image download

Add inputs to resize the image when downloading the png version of it.

madcampos committed:
feat: update readme

madcampos committed:
chore: update packages

madcampos committed:
chore: update tools config

madcampos committed:
feat: make downloadable image more generic

madcampos committed:
chore: add TODO

madcampos committed:
feat: add base for editable graphic

madcampos committed:
chore: ignore tsc errors

madcampos committed:
feat: update first image to use new format

madcampos committed:
chore: add TODO

madcampos committed:
feat: update project sctructure

madcampos committed:
feat: update downloadable image element

madcampos committed:
feat: update color box

madcampos committed:
fix: typos

madcampos committed:
fix: file path

madcampos committed:
fix: cleanup element

madcampos committed:
feat: add base implementation for editable image

madcampos committed:
feat: add working image with changing bg

madcampos committed:
chore: add TODO for components

madcampos committed:
feat: write all sections of style guide

madcampos committed:
feat: add volunteer badge

madcampos committed:
feat: remove custom scrollbars

madcampos committed:
feat: implement events for editable image

madcampos committed:
feat: update image components

madcampos committed:
fix: use slotted selector for css

madcampos committed:
feat: update styleguide

madcampos committed:
feat: rework images height to avoid cumulative layout shift

madcampos committed:
feat: implement dynamic TOC and move it to a component

madcampos committed:
feat: update color-box component

madcampos committed:
chore: delete leftover file

madcampos committed:
chore: cleanup assets

madcampos committed:
feat: add basic font box

madcampos committed:
fix: add scroll to link targets

madcampos committed:
fix: add missing color names

madcampos committed:
chore: remove redundant type jsdoc annotation

madcampos committed:
feat: implement text fitting logic

madcampos committed:
chore: update stylelint rule

madcampos committed:
fix: resolve dynamic update of text fitting attribute

madcampos committed:
fix: update CSS current color casing

madcampos committed:
feat: publish new version

madcampos committed:
feat: fix css for forms

madcampos committed:
feat: update build

usrrname commented:
It's insane how similar our commit habits are... ie. every commit a story line and prefix via commitizen-convention lmao

usrrname commented on .stylelintrc:
```diff
@@ -0,0 +1,103 @@
+		"value-keyword-case": ["lower", { "camelCaseSvgKeywords": true }],
+		"value-no-vendor-prefix": [true, { "ignoreValues": ["box", "grab", "no-grab"] }]
+	}
+}
```
I'm usually not a fan of excessive lint rules but I imagine it's strict because you're generating and saving files

madcampos replied:
We can remove stylelint for sure. It is a linter for CSS to have a more "normalized" code. Most of the rules are not too restrictive though, like, there is no rule for order of properties or things like that.

I'm just a weird person that doesn't like other people's configs and I like to be explicit in all of the options, so lots of them are turned off though. If it is better, we can cleanup the rules to be more concise or use someone else's config, no problem.

usrrname commented on vite.config.ts:
```diff
@@ -0,0 +1,106 @@
+
+	const sslOptions = mode === 'production'
+		? false
+		: {
```
I dunno if it's because I'm on a M2 and I know nothing of mkcert other than having to generate them at work also. For somereason my computer insisted on creating .pem files, regardless by changing these lines to: 

```
 {
          cert: readFileSync("./certutil.pem"),
          key: readFileSync("./certutil-key.pem"),
        }
```

... I was able to get it running!

usrrname commented on src/components/downloadable-image/index.ts:
```diff
@@ -0,0 +1,304 @@
+ * @slot - The image to be displayed and enhanced.
+ * @element downloadable-image
+ */
+export class DownloadableImage extends HTMLElement {
```
Even these are web components... my god this is masochistic

usrrname commented on src/service-worker.ts:
```diff
@@ -0,0 +1,25 @@
+import type { VitePWAOptions } from 'vite-plugin-pwa';
+
```
This is thoroughly fascinating as I never used web/service workers before.

madcampos replied:
Those are basically two rules that we pass to [workbox](https://developer.chrome.com/docs/workbox/).
1. Cache for 30 days every response for the app, so all internal assets and files will be cached.
2. Don't cache and pass straight to the network all external requests.

Service workers are cool and all but it is painful to configure one by hand and deal with every part of it, most people use workbox because it is configurable and easy.

usrrname reviewed (approved):
A magnificent contribution! 

Non-blocking edit: I wouldn't require all the history preamble on this page, after all people likely wanna use this to get to the branding guide or social media card/ badge generator.

Seems a merge conflict remains, so probably just needs a rebase. 

Feel free to make a branch linking to this style guide as well from torontojs.com

madcampos commented on README.md:
````diff
@@ -0,0 +1,54 @@
+
+You can run the following command:
+```shell
+$ mkcert -key-file certs/server.key -cert-file certs/server.crt localhost
````
@usrrname for generating the certificates run this command on the project home

usrrname replied:
yep that's what I did, I think it might just be a M2 thing

madcampos committed:
chore: add nolyfill

madcampos committed:
fix: resolve typo on color name attribute

madcampos committed:
fix: revert from using popover to a dialog element for TOC

Solves an issue where browsers that don't support popover attributes still display the TOC.
Also moves the internals of the component to be in the light DOM rather than on the shadow DOM.
This better respect font sizes and other CSS properties.

madcampos committed:
feat: update build

usrrname committed:
Merge branch 'main' into feat/update-style-guide

usrrname commented:
:shipit:
